### PR TITLE
parse opening hours from GISCorps dataset using homemade library

### PR DESCRIPTION
**Pull request by MoralCode**
| Key Details |
|-|
State: US
Site: giscorps_vaccine_providers

## Notes
uses a custom library that i made (https://github.com/MoralCode/jsonify-opening-hours) to parse the arbitrary-looking strings in the "operhours" of the GISCorps data into the dict that is accepted by the schema.

This library doesnt handle all cases, but can handle the most common ones.
In the event that the library cannot handle a particular input, that value is stored and added to the end of the notes field so it at least exists somewhere.

## Before Opening a PR
- [ X] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [X ] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py [hunk 1]
```diff
@@ -202,7 +227,9 @@ def _get_normalized_location(site: dict, timestamp: str) -> schema.NormalizedLoc
         contact=_get_contacts(site),
         languages=None,
         opening_dates=None,
-        opening_hours=None,  # TODO: the format for this probably needs some mega-parsing as it looks like this -> "operhours": "Monday - Friday 8:00 am - 2:00 pm Saturdays 9:00 am - 12:00 pm",
+        opening_hours=_get_opening_hours(
+            site
+        ),  # TODO: the format for this probably needs some mega-parsing as it looks like this -> "operhours": "Monday - Friday 8:00 am - 2:00 pm Saturdays 9:00 am - 12:00 pm",
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:232]: Is this comment still needed?

**Review by eliblock — CHANGES_REQUESTED** [review]
Please resolve the merge conflicts so we can run CI and test this!

I'm excited to see that you've been working on this in https://github.com/MoralCode/jsonify-opening-hours. In your comment, you mention that it's scoped to the GISCorps format - could you expand on what you mean (in other words, could this library be used elsewhere)?

In addition, could you share a bit about how you build and test the library?

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py (hunk 1 above)
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:232]: no, not anymore, i just did not notice that when implementing this

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py [hunk 2]
```diff
@@ -7,6 +7,11 @@
 import sys
 from typing import List, Optional
 
+try:
+    from opening_hours import OpeningHours
+except Exception:
+    from parse_opening_hours import OpeningHours
```
[the remark this answers is not in the document]
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:13]: will fix, was just being indecisive about releasing breaking changes.

File: pyproject.toml [hunk 3]
```diff
@@ -46,6 +46,7 @@ ndjson = "^0.3.1"
 orjson = "^3.5.2"
 diskcache = "^5.2.1"
 placekey = "^0.0.11"
+jsonify-opening-hours = "^0.3.1"
```
[the remark this answers is not in the document]
Comment (MoralCode) [reply pyproject.toml]: Np!

**Comment by MoralCode** [comment]
>  In your comment, you mention that it's scoped to the GISCorps format - could you expand on what you mean (in other words, could this library be used elsewhere)?

Yeah my initial comment in the PR message may have been confusing. What I meant was that the "operhours" field, which is part of the GISCorps data schema, contains an arbitrary string that represents opening hours. I presume this string was intended for display on a website and not machine parsing which is probably why its so inconsistently formatted.

The whole reason i made this library is because i wasnt able to find one and it seemed like being able to parse the most common opening hours style strings would be something that could also be useful to others as a standalone library.


> In addition, could you share a bit about how you build and test the library?

Since the primary reason i made the library was for this usecase, I pretty much started with some of the simpler strings from the GISCorps dataset to get a basic version and started buinding from there. I also tried to be really thorough by making unit tests that covered every new string format as I went so the unit tests should be very representative of what will work. Not sure if thats as much info as you wanted on that, but if it helps, I also wrote a [blog post](https://www.adriancedwards.com/blog/2021-05-06-fighting-covid-with-open-source) about this library in case that has any more detail in it that I missed.

**Review by bryanculbertson — APPROVED** [review]

**Comment by bryanculbertson** [comment]
I am going to approve and mere to unblock this. It looks like a good approach to me.

**Pull request merged**